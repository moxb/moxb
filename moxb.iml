<?xml version="1.0" encoding="UTF-8"?>
<module type="WEB_MODULE" version="4">
  <component name="NewModuleRootManager" inherit-compiler-output="true">
    <exclude-output />
    <content url="file://$MODULE_DIR$">
      <sourceFolder url="file://$MODULE_DIR$/packages/antd/src" isTestSource="false" />
      <sourceFolder url="file://$MODULE_DIR$/packages/meteor/src" isTestSource="false" />
      <sourceFolder url="file://$MODULE_DIR$/packages/moxb/src" isTestSource="false" />
      <sourceFolder url="file://$MODULE_DIR$/packages/react-html/src" isTestSource="false" />
      <sourceFolder url="file://$MODULE_DIR$/examples/src" isTestSource="false" />
      <excludeFolder url="file://$MODULE_DIR$/packages/moxb/dist" />
      <excludeFolder url="file://$MODULE_DIR$/packages/meteor/dist" />
      <excludeFolder url="file://$MODULE_DIR$/packages/react-html/dist" />
      <excludeFolder url="file://$MODULE_DIR$/packages/antd/dist" />
      <excludeFolder url="file://$MODULE_DIR$/admin/node-installation" />
      <excludeFolder url="file://$MODULE_DIR$/admin/bin-tools" />
      <excludeFolder url="file://$MODULE_DIR$/examples/dist" />
      <excludeFolder url="file://$MODULE_DIR$/examples/.cache" />
      <excludeFolder url="file://$MODULE_DIR$/coverage" />
    </content>
    <orderEntry type="inheritedJdk" />
    <orderEntry type="sourceFolder" forTests="false" />
  </component>
</module>