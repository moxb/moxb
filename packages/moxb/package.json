{
  "name": "@moxb/moxb",
  "version": "0.2.0-beta.59",
  "description": "MobX data binding",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "main": "dist",
  "typings": "dist/",
  "files": [
    "dist/**/*"
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/moxb/moxb.git"
  },
  "author": "moxb (moxb.org)",
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/moxb/moxb/issues"
  },
  "homepage": "https://github.com/moxb/moxb#readme",
  "keywords": [
    "mobx",
    "UI"
  ],
  "jest": {
    "moduleFileExtensions": [
      "ts",
      "tsx",
      "js",
      "jsx",
      "json",
      "node"
    ],
    "transform": {
      "^.+\\.(ts|tsx)$": "ts-jest"
    },
    "testEnvironment": "node",
    "testMatch": [
      "**/src/**/__tests__/*.(ts|tsx)"
    ],
    "modulePaths": [
      "<rootDir>/peer-dependencies/node_modules"
    ]
  },
  "peerDependencies": {
    "mobx": "^5.15.6",
    "tslib": "^2.0.1"
  },
  "devDependencies": {
    "@types/atob": "2.1.2",
    "@types/jest": "26.0.13",
    "@types/lodash.flatten": "4.4.6",
    "@types/urijs": "1.19.11",
    "mobx": "5.13.1",
    "tslib": "1.10.0"
  },
  "dependencies": {
    "btoa-lite": "1.0.0",
    "history": "5.0.0",
    "lodash.flatten": "4.4.0",
    "moment": "2.29.1",
    "urijs": "1.19.2"
  }
}
