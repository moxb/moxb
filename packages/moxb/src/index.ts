export * from './impl';
export * from './types';
export * from './routing';
