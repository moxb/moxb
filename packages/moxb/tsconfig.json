{
    "compilerOptions": {
        "module": "commonjs",
        "target": "ES5",
        "jsx": "react",
        "lib": ["dom","es6","es2015","es2017", "dom.iterable","scripthost"],
        "inlineSourceMap": true,
        "importHelpers":true,
        "experimentalDecorators": true,
        "noImplicitAny": true,
        "noImplicitThis": true,
        "strict": true,
        "outDir": "./dist",
        "declaration": true,
        "skipLibCheck": true,
        "strictNullChecks": true,
        "removeComments": false,
        "noUnusedLocals": true,
        "noUnusedParameters": true,
        "allowUnreachableCode": false,
        "baseUrl": "./src",
        "paths": {
            "*": [
                "*",
                "../../../node_modules/*"
            ]
        }
    },
    "exclude": [
        "./node_modules",
        "./dist"
    ]
}
