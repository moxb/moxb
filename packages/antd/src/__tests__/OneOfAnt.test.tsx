import { OneOfImpl } from '@moxb/moxb';
import { shallow } from 'enzyme';
import * as React from 'react';
import {
    OneOfAnt,
    OneOfFormAnt,
    OneOfSelectAnt,
    OneOfSelectFormAnt,
    OneOfDropDownAnt,
    OneOfDropDownFormAnt,
} from '../OneOfAnt';
import { shallowMoxbToJson } from './enzymeHelper';

const allChoices = [{ label: 'Banana', value: 'b' }, { label: 'Apples', value: 'a' }, { label: 'Peaches', value: 'p' }];

describe('OneOfAnt', function() {
    it('should render a radio button group by default', function() {
        const operation = new OneOfImpl({
            id: 'id.testRadioOfOne',
            label: 'Select one of',
            placeholder: '...',
            choices: () => allChoices,
        });
        expect(shallowMoxbToJson(shallow(<OneOfAnt operation={operation} />))).toMatchSnapshot();
    });

    it('should return null if invisible', function() {
        const operation = new OneOfImpl({
            id: 'id.testRadioOfOne',
            label: 'Select one of',
            placeholder: '...',
            choices: () => allChoices,
            invisible: () => true,
        });
        expect(shallow(<OneOfAnt operation={operation} />).type()).toBeNull();
    });
});

describe('OneOfFormAnt', function() {
    it('should render a radio button group with form by default', function() {
        const operation = new OneOfImpl({
            id: 'id.testRadioOfOne',
            label: 'Select one of',
            placeholder: '...',
            choices: () => allChoices,
        });
        expect(shallowMoxbToJson(shallow(<OneOfFormAnt operation={operation} />))).toMatchSnapshot();
    });

    it('should return null if invisible', function() {
        const operation = new OneOfImpl({
            id: 'id.testRadioOfOne',
            label: 'Select one of',
            placeholder: '...',
            choices: () => allChoices,
            invisible: () => true,
        });
        expect(shallow(<OneOfFormAnt operation={operation} />).type()).toBeNull();
    });
});

describe('OneOfSelectAnt', function() {
    it('should render a select control', function() {
        const operation = new OneOfImpl({
            id: 'id.testRadioOfOne',
            label: 'Select one of',
            placeholder: '...',
            choices: () => allChoices,
        });
        expect(shallowMoxbToJson(shallow(<OneOfSelectAnt operation={operation} />))).toMatchSnapshot();
    });

    it('should return null if invisible', function() {
        const operation = new OneOfImpl({
            id: 'id.testRadioOfOne',
            label: 'Select one of',
            placeholder: '...',
            choices: () => allChoices,
            invisible: () => true,
        });
        expect(shallow(<OneOfSelectAnt operation={operation} />).type()).toBeNull();
    });
});

describe('OneOfSelectFormAnt', function() {
    it('should render a select control with form', function() {
        const operation = new OneOfImpl({
            id: 'id.testRadioOfOne',
            label: 'Select one of',
            placeholder: '...',
            choices: () => allChoices,
        });
        expect(shallowMoxbToJson(shallow(<OneOfSelectFormAnt operation={operation} />))).toMatchSnapshot();
    });

    it('should return null if invisible', function() {
        const operation = new OneOfImpl({
            id: 'id.testRadioOfOne',
            label: 'Select one of',
            placeholder: '...',
            choices: () => allChoices,
            invisible: () => true,
        });
        expect(shallow(<OneOfSelectFormAnt operation={operation} />).type()).toBeNull();
    });
});

describe('OneOfDropDownAnt', function() {
    it('should render a dropdown control', function() {
        const operation = new OneOfImpl({
            id: 'id.testRadioOfOne',
            label: 'Select one of',
            placeholder: '...',
            choices: () => allChoices,
        });
        expect(shallowMoxbToJson(shallow(<OneOfDropDownAnt operation={operation} />))).toMatchSnapshot();
    });

    it('should return null if invisible', function() {
        const operation = new OneOfImpl({
            id: 'id.testRadioOfOne',
            label: 'Select one of',
            placeholder: '...',
            choices: () => allChoices,
            invisible: () => true,
        });
        expect(shallow(<OneOfDropDownAnt operation={operation} />).type()).toBeNull();
    });
});

describe('OneOfDropDownFormAnt', function() {
    it('should render a dropdown control with form', function() {
        const operation = new OneOfImpl({
            id: 'id.testDropDownOfOne',
            label: 'Select one of',
            placeholder: '...',
            choices: () => allChoices,
        });
        expect(shallowMoxbToJson(shallow(<OneOfDropDownFormAnt operation={operation} />))).toMatchSnapshot();
    });

    it('should return null if invisible', function() {
        const operation = new OneOfImpl({
            id: 'id.testDropDownOfOne',
            label: 'Select one of',
            placeholder: '...',
            choices: () => allChoices,
            invisible: () => true,
        });
        expect(shallow(<OneOfDropDownFormAnt operation={operation} />).type()).toBeNull();
    });
});
