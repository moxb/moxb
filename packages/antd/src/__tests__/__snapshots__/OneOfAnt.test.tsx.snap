// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OneOfAnt should render a radio button group by default 1`] = `
<RadioGroup
  buttonStyle="outline"
  data-testid="id-test_radio_of_one"
  disabled={false}
  id="id-test_radio_of_one"
  label="Select one of"
  onChange={[Function]}
  readOnly={false}
>
  <Radio
    data-testid="id-test_radio_of_one-b"
    key="b"
    type="radio"
    value="b"
  >
    Banana
  </Radio>
  <Radio
    data-testid="id-test_radio_of_one-a"
    key="a"
    type="radio"
    value="a"
  >
    Apples
  </Radio>
  <Radio
    data-testid="id-test_radio_of_one-p"
    key="p"
    type="radio"
    value="p"
  >
    Peaches
  </Radio>
</RadioGroup>
`;

exports[`OneOfDropDownAnt should render a dropdown control 1`] = `
<Dropdown
  data-testid="id-test_drop_down_of_one"
  disabled={false}
  label="Select one of"
  mouseEnterDelay={0.15}
  mouseLeaveDelay={0.1}
  overlay={
    <Menu
      data-testid="id.testDropDownOfOne-menu"
    >
      <MenuItem
        data-testid="id-test_drop_down_of_one-menu-b"
        onClick={[Function]}
      >
        Banana
      </MenuItem>
      <MenuItem
        data-testid="id-test_drop_down_of_one-menu-a"
        onClick={[Function]}
      >
        Apples
      </MenuItem>
      <MenuItem
        data-testid="id-test_drop_down_of_one-menu-p"
        onClick={[Function]}
      >
        Peaches
      </MenuItem>
    </Menu>
  }
  trigger={
    Array [
      "click",
    ]
  }
>
  <Button
    block={false}
    ghost={false}
    htmlType="button"
    id="id-test_drop_down_of_one"
    loading={false}
    style={
      Object {
        "alignItems": "center",
        "display": "inline-flex",
        "justifyContent": "space-between",
      }
    }
  >
    <span
      style={
        Object {
          "color": "#aaaaaa",
        }
      }
    >
      ...
    </span>

    <ForwardRef(DownOutlined) />
  </Button>
</Dropdown>
`;

exports[`OneOfDropDownFormAnt should render a dropdown control with form 1`] = `
<FormItemAnt
  operation="bound OneOfImpl"
>
  <OneOfDropDownAnt
    disabled={false}
    operation="bound OneOfImpl"
    readOnly={false}
  />
</FormItemAnt>
`;

exports[`OneOfFormAnt should render a radio button group with form by default 1`] = `
<FormItemAnt
  operation="bound OneOfImpl"
>
  <OneOfAnt
    disabled={false}
    id="id-test_radio_of_one"
    operation="bound OneOfImpl"
    readOnly={false}
  />
</FormItemAnt>
`;

exports[`OneOfSelectAnt should render a select control 1`] = `
<Select
  bordered={true}
  choiceTransitionName="zoom"
  data-testid="id.testSelectOfOne"
  disabled={false}
  id="id-test_select_of_one"
  label="Select one of"
  mode="tags"
  onChange={[Function]}
  placeholder="..."
  transitionName="slide-up"
>
  <Option
    data-testid="id-test_select_of_one-b"
    key="b"
    value="b"
  >
    Banana
  </Option>
  <Option
    data-testid="id-test_select_of_one-a"
    key="a"
    value="a"
  >
    Apples
  </Option>
  <Option
    data-testid="id-test_select_of_one-p"
    key="p"
    value="p"
  >
    Peaches
  </Option>
</Select>
`;

exports[`OneOfSelectFormAnt should render a select control with form 1`] = `
<FormItemAnt
  operation="bound OneOfImpl"
>
  <OneOfSelectAnt
    disabled={false}
    id="id-test_select_of_one"
    operation="bound OneOfImpl"
    readOnly={false}
  />
</FormItemAnt>
`;
